### PR TITLE
[REVIEW] Add random allocation benchmark

### PR DESCRIPTION
**Pull request by jrhemstad**
Extracts the benchmark from https://github.com/rapidsai/rmm/pull/162

@harrism when ran as-is, I was surprised to see CNMEM being slower than CUDA? Is this expected?

```
--------------------------------------------------------------------
Benchmark                          Time             CPU   Iterations
--------------------------------------------------------------------
BM_RandomAllocationsCUDA        7595 ms         6422 ms            1
BM_RandomAllocationsCnmem      14571 ms        14544 ms            1

```

### CODE REVIEW

File: benchmarks/random_allocations/random_allocations.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+// TODO figure out how to map a normal distribution to integers between 1 and max_allocation_size
+/*void normal_random_allocations(rmm::mr::device_memory_resource& mr,
+                                size_t num_allocations = 1000,
+                                size_t mean_allocation_size = 500, // in MiB
+                                size_t stddev_allocation_size = 500, // in MiB
+                                size_t max_usage = 8 << 20,
+                                cudaStream_t stream) {
+  std::normal_distribution<std::size_t> size_distribution(, max_allocation_size * size_mb);
```
Comment (jrhemstad) [inline benchmarks/random_allocations/random_allocations.cpp]: Couldn't we just cast the `double` to an integer? Technically that would make the distribution not a statistically perfect "normal" distribution, but I think that's okay for our purposes here.

Comment (harrism) [reply benchmarks/random_allocations/random_allocations.cpp]: And clamp to 0, max_allocation_size?

Comment (harrism) [reply benchmarks/random_allocations/random_allocations.cpp]: I think for our purposes we may not even need this test.

File: benchmarks/test/test_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#include <benchmark/benchmark.h>
```
Comment (harrism) [inline benchmarks/test/test_benchmark.cpp]: Don't really need this benchmark.

**Comment by harrism** [comment]
cnmem being slower is not expected. But it may be due to fragmentation since there are a large number of relatively small allocations with the default settings. If you reduce #iterations by 10x perf of cnmem goes up by 100x! (This is why the benchmark is needed!)

**Pull request merged**